packages/zoe/src/contractFacet/zcfMint.js

```diff
@@ -0,0 +1,182 @@
+
+// helpers for the code shared between MakeZCFMint and RegisterZCFMint
+
+export const makeZCFMintFactory = async (
+  zcfBaggage,
+  recordIssuer,
+  getAssetKindByBrand,
+  makeEmptySeatKit,
+  reallocateForZCFMint,
+) => {
+  // The set of baggages for zcfMints
+  const zcfMintBaggageSet = provideDurableSetStore(zcfBaggage, 'baggageSet');
+
+  /**
+   * retrieve the state of the zcfMint from the baggage, and create a durable
+   * singleton reflecting that state.
+   *
+   * @param {import('@agoric/vat-data').Baggage} zcfMintBaggage
+   */
+  const provideDurableZcfMint = async zcfMintBaggage => {
+    const keyword = zcfMintBaggage.get('keyword');
+    const zoeMint = zcfMintBaggage.get('zoeMint');
+    const {
+      brand: mintyBrand,
+      issuer: mintyIssuer,
+      displayInfo: mintyDisplayInfo,
+    } = await E(zoeMint).getIssuerRecord();
+    // AWAIT
+    const mintyIssuerRecord = makeIssuerRecord(
+      mintyBrand,
+      mintyIssuer,
+      mintyDisplayInfo,
+    );
+    recordIssuer(keyword, mintyIssuerRecord);
+
+    const empty = AmountMath.makeEmpty(mintyBrand, mintyDisplayInfo.assetKind);
+    const add = (total, amountToAdd) => {
+      return AmountMath.add(total, amountToAdd, mintyBrand);
+    };
+
+    return vivifySingleton(
+      zcfMintBaggage,
+      'zcfMint',
+      /** @type {ZCFMint} */
+      {
+        getIssuerRecord: () => {
+          return mintyIssuerRecord;
+        },
+        mintGains: (gains, zcfSeat = undefined) => {
+          gains = coerceAmountKeywordRecord(gains, getAssetKindByBrand);
+          if (zcfSeat === undefined) {
+            zcfSeat = makeEmptySeatKit().zcfSeat;
+          }
+          const totalToMint = Object.values(gains).reduce(add, empty);
+          assert(
+            // @ts-expect-error It's non-null
+            !zcfSeat.hasExited(),
+            `zcfSeat must be active to mint gains for the zcfSeat`,
+          );
+          const allocationPlusGains = addToAllocation(
+            // @ts-expect-error It's non-null
+            zcfSeat.getCurrentAllocation(),
+            gains,
+          );
+
+          // Increment the stagedAllocation if it exists so that the
+          // stagedAllocation is kept up to the currentAllocation
+          // @ts-expect-error It's non-null
+          if (zcfSeat.hasStagedAllocation()) {
+            // @ts-expect-error It's non-null
+            zcfSeat.incrementBy(gains);
+          }
+
+          // Offer safety should never be able to be violated here, as
+          // we are adding assets. However, we keep this check so that
+          // all reallocations are covered by offer safety checks, and
+          // that any bug within Zoe that may affect this is caught.
+          assert(
+            // @ts-expect-error It's non-null
+            zcfSeat.isOfferSafe(allocationPlusGains),
+            `The allocation after minting gains ${allocationPlusGains} for the zcfSeat was not offer safe`,
```

Comment:
This potentially leaks actual amounts.
```suggestion
            X`The allocation after minting gains ${allocationPlusGains} for the zcfSeat was not offer safe`,
```

Reply:
and on line 126.